appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/visitor/OpenApiWalker.java

```diff
@@ -182,17 +184,33 @@ public void accept(ApiVisitor visitor) {
 
             processAnnotation(clazz, annotationClass, annotationFunction, altClass, altFunction,
                     new OpenApiContext(api, getResourcePath(clazz)));
-
+            List<Field> fields = new ArrayList<>();
             for (Field field : clazz.getDeclaredFields()) {
-                processAnnotation(field, annotationClass, annotationFunction, altClass, altFunction,
```

Comment:
I believe we discussed removing this? This class would be best kept generic, moving the implementation specific logic into the handleParam methods.